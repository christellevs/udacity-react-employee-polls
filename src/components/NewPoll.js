import { useState } from "react";
import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import { handleAddQuestion } from "../actions/questions.js";

const NewPoll = ({ dispatch }) => {
  const navigate = useNavigate();
  const [optionOneText, setOptionOneText] = useState("");
  const [optionTwoText, setOptionTwoText] = useState("");

  const onChangeOptionOne = (e) => {
    setOptionOneText(e.target.value);
  };

  const onChangeOptionTwo = (e) => {
    setOptionTwoText(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    dispatch(handleAddQuestion(optionOneText, optionTwoText));

    setOptionOneText("");
    setOptionTwoText("");
    navigate("/");
  };

  return (
    <div>
      <h3>Would You Rather?</h3>
      <h6>Create Your Own Poll</h6>

      <form className="new-poll" onSubmit={handleSubmit}>
        <div>
          <input
            type="text"
            value={optionOneText}
            placeholder="Option One..."
            onChange={onChangeOptionOne}
          />
          <input
            type="text"
            value={optionTwoText}
            placeholder="Option Two..."
            onChange={onChangeOptionTwo}
          />
        </div>
        <div>
          <button
            className="btn"
            type="submit"
            disabled={optionOneText === "" || optionTwoText === ""}
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  );
};

export default connect()(NewPoll);
