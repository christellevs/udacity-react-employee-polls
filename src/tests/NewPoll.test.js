import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { store } from "../store.js";
import NewPoll from "../components/NewPoll";

describe("NewPoll", () => {
  it("will match snapshot", () => {
    var component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPoll />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
