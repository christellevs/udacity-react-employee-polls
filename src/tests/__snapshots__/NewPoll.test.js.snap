// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2>
          Would You Rather?
        </h2>
        <h4>
          Create Your Own Poll
        </h4>
        <form
          class="new-poll"
        >
          <div
            class="new-poll-inputs-container"
          >
            <input
              class="new-poll-input"
              placeholder="Option One..."
              type="text"
              value=""
            />
            <input
              class="new-poll-input"
              placeholder="Option Two..."
              type="text"
              value=""
            />
          </div>
          <div>
            <button
              class="btn btn-new-poll"
              disabled=""
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2>
        Would You Rather?
      </h2>
      <h4>
        Create Your Own Poll
      </h4>
      <form
        class="new-poll"
      >
        <div
          class="new-poll-inputs-container"
        >
          <input
            class="new-poll-input"
            placeholder="Option One..."
            type="text"
            value=""
          />
          <input
            class="new-poll-input"
            placeholder="Option Two..."
            type="text"
            value=""
          />
        </div>
        <div>
          <button
            class="btn btn-new-poll"
            disabled=""
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
